server/tests/helpers/user_helper.js

```diff
@@ -0,0 +1,21 @@
+import {fixtures, build} from '../factories'
+
+function buildUserAndGroups(){
+  return new Promise((resolve, reject) => {
+    let generated_values = {}
+    const b1 = build('group', fixtures.group())
+    const b2 = build('group', fixtures.group())
+    Promise.all([b1, b2])
+      .then( (values) => {
+        generated_values.groups = values;
+        let groupNames = values.map( g => { return g.name });
+        build('user', fixtures.user({groups: groupNames}))
```

Comment:
You should be able to add a `return` on this line and call the `.then()` in the highest level block. 

```
.then(...
  return build('user', fixtures.user({groups: groupNames}))
).then((user) => {
  generated_values.user = u;
  return generated_values;
});
```